addons/fullscreen/userscript.js

```diff
@@ -31,6 +31,58 @@ export default async function ({ addon, console }) {
     }
   }
 
+  // The "phantom header" is a small strip at the top of the page that
+  // brings the header into view when hovered.
+  async function updatePhantomHeader() {
+    if (
+      !addon.self.disabled &&
+      addon.tab.redux.state.scratchGui.mode.isFullScreen &&
+      addon.settings.get("hideToolbar") &&
+      addon.settings.get("hoverToolbar")
+    ) {
+      const canvas = await addon.tab.waitForElement('[class*="stage_full-screen"] canvas');
+      const header = await addon.tab.waitForElement('[class^="stage-header_stage-header-wrapper"]');
+      const phantom = header.parentElement.appendChild(document.createElement("div"));
+      phantom.classList.add("phantom-header");
+
+      // Make the header a child of the phantom, so that mouseleave will trigger when the
+      // mouse leaves the header OR the phantom header.
+      phantom.appendChild(header);
+
```

Comment:
Try this, it might solve the problem of the mouse slipping past the phantom header when the browser tab isn't full screen.
```suggestion

      // Listen for when the mouse moves above the page (helps when not in browser full screen mode)
      document.addEventListener("mouseleave", (e) => {
        if (e.clientY < 8) {
          header.classList.add("stage-header-hover");
        }
      });
      // and for when the mouse re-enters the page
      document.addEventListener("mouseenter", () => {
        header.classList.remove("stage-header-hover");
      });

```
Code snippet tested.

Reply:
Could also detect if the mouse is within the document when it leaves the phantom header (if it isn't, then don't remove the `stage-header-hover` class) if possible, in addition to detecting when the mouse re-enters the document.

Reply:
I think the first option would be easier to implement.